### PR TITLE
remove builder event types

### PR DESCRIPTION
**Pull request by imabdulbasit**
This PR:

- removes BuilderEvent and BuilderEventTypes
- adds a filter parameter to EventStreamer

### CODE REVIEW

**Review by elliedavidson — COMMENTED** [review]
LGTM but I'll defer approval to someone more familiar with this repo.

File: src/events_source.rs [hunk 1]
```diff
@@ -1,64 +1,22 @@
...
-                }
-                _ => BuilderEventType::Unknown,
-            },
-        }
-    }
-}
-
 #[derive(Clone, Debug, Serialize, Deserialize)]
 #[serde(bound(deserialize = "Types: NodeType"))]
 pub enum BuilderEventType<Types: NodeType> {
```
[the remark this answers is not in the document]
Comment (imabdulbasit) [reply src/events_source.rs]: removed

File: src/events_source.rs [hunk 2]
```diff
@@ -1,76 +1,22 @@
...
-
 #[async_trait]
 pub trait EventsSource<Types>
 where
     Types: NodeType,
 {
     type EventStream: Stream<Item = Arc<Event<Types>>> + Unpin + Send + 'static;
-    async fn get_event_stream(&self) -> Self::EventStream;
-
-    async fn subscribe_events(&self) -> BoxStream<'static, Arc<Event<Types>>> {
```
Comment (imabdulbasit) [inline src/events_source.rs]: looked unnecessary

**Review by jbearer — APPROVED** [review]

**Comment by move47** [comment]
Hi, Earlier we defined BuilderEvents to suppress the unnecessary stuff embedded in HotShot events which builder don't care about and it used to affect builder performance. For e.g . we removed something from Decide event here: https://github.com/EspressoSystems/hotshot-events-service/pull/53. I am not sure whether that went away with this change.

**Pull request merged**